src/common/hotbar-store.ts

```diff
@@ -34,32 +38,98 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
     });
   }
 
+  get activeHotbarId() {
+    return this._activeHotbarId;
+  }
+
+  set activeHotbarId(id: string) {
+    if (this.getById(id)) {
+      this._activeHotbarId = id;
+    }
+  }
+
+  get activeHotbarIndex() {
+    return this.hotbars.findIndex((hotbar) => hotbar.id === this.activeHotbarId);
+  }
+
   @action protected async fromStore(data: Partial<HotbarStoreModel> = {}) {
     if (data.hotbars?.length === 0) {
       this.hotbars = [{
-        name: "default",
+        id: uuid.v4(),
+        name: "Default",
         items: []
       }];
     } else {
       this.hotbars = data.hotbars;
     }
+
+    if (data.activeHotbarId) {
+      if (this.getById(data.activeHotbarId)) {
+        this.activeHotbarId = data.activeHotbarId;
+      }
+    }
+
+    if (!this.activeHotbarId) {
+      this.activeHotbarId = this.hotbars[0].id;
+    }
+  }
+
+  getActive() {
+    return this.getById(this.activeHotbarId);
   }
 
   getByName(name: string) {
     return this.hotbars.find((hotbar) => hotbar.name === name);
   }
 
-  add(hotbar: Hotbar) {
-    this.hotbars.push(hotbar);
+  getById(id: string) {
+    return this.hotbars.find((hotbar) => hotbar.id === id);
+  }
+
+  add(hotbar: Partial<Hotbar>) {
+    hotbar.id = uuid.v4();
+
+    if (!hotbar.items) {
+      hotbar.items = [];
+    }
+
+    this.hotbars.push(hotbar as Hotbar);
+
+    return hotbar as Hotbar;
   }
```

Comment:
You don't need the casts if you create the entry after deconstruction. Finally, except for `render*()` functions I think that the return value should be explicitly written out so that function definitions are fixed points. This also should help when changing the implementation of a function because typescript will fail to compile if you don't satisfy the contract.

Plus it seems like you want the `name` to still be required:

```suggestion
interface HotbarCreateOptions extends Hotbar {
  id?: string;
  items?: HotbarItem[];
}

  add(rawHotbar: HotbarCreateOptions): Hotbar {
  	const {
  	  id = uuid.v4(), // initialize to a new ID if not provided
  	  items = [], // initialize to empty array if not provided
  	  name,
  	} = rawHotbar;
  	const hotbar = { id, items, name };

    this.hotbars.push(hotbar);

    return hotbar;
  }
```